src/blog/2023/06/Flowforge-as-a-No-Code-Ethernet_IP-to-Profinet-Protocol-Converter.md

```diff
@@ -0,0 +1,208 @@
+---
+title: "Flowforge as a No-Code Ethernet/IP to Profinet Protocol Converter"
+subtitle: Beginner tutorial for using node-red as free industrial protocol converter
+description: step-by-step guide for using flowforge as an industrial protocol converter
+date: 2023-06-19
+authors: ["richard-meyer"]
+image:
+tags:
+  - posts
+  - how-to
+---
+
+
+Frequently in industrial automation, there's a need for two devices that use different protocols to communicate with each other, requiring protocol conversion.  
+In this article, we present a mock scenario where flowforge is used to enable an Allen Bradley PLC, which uses ethernet/IP, to communicate with a Siemens PLC, which uses Profinet, using a no-code solution. This example is geared toward beginners and assumes that the end-user knows how to use PLCs, but may be using flowforge/node-red for the first time.
+
+<!--more-->
+
+# Premise
+
+![1](./images/ethip-to-profinet/e-to-p-1.png)
+
+Figure 1 above shows the layout of a mock production facility. Inside our facility, operations suggested adding stacklights as an extra visual aid for operators to get a quick status of its 4 conveyor lines, avoiding the need to constantly monitor the HMI/SCADA displays.  
+Engineering has suggested adding a siemens S7 1200 PLC with an IO link connection to 4 stacklights, with each line PLC sending basic status information to the stacklight PLC to control the stacklight outputs.  
+Line 1-3 PLCs are Siemens-based, and can communicate with the stacklight PLC natively over Profinet. But line 4 is an Allen Bradley PLC that uses ethernet/IP, and can't communicate with the stacklight PLC without some form of protocol conversion.  
+Traditionally, we'd use protocol gateway hardware, like anybus or red lion, to convert ethernet/IP to profinet.  
+But for this application, we will instead use flowforge, a pure software-based approach, to convert ethernet/IP to Profinet. Let's walk through the process.
```

Comment:
Before "Let's walk you through the process" it would be great to have 1 sentence to say why a software based approach is wanted

```suggestion
But for this application, we will instead use FlowForge, a pure software-based approach, to convert ethernet/IP to Profinet. Let's walk through the process.
```